packages/framework-types/src/searcher.ts

```diff
@@ -77,8 +77,14 @@ export class Searcher<TObject> {
 
   public async searchOne(): Promise<TObject> {
     // TODO: If there is only an ID filter with one value, this should call to `findById`
-    const searchResult = await this.search()
-    return Array.isArray(searchResult) ? searchResult[0] : searchResult.items[0]
+    const searchResult = await this.searcherFunction(
+      this.objectClass.name,
+      this.filters,
+      1, // Forces limit 1
+      this._afterCursor,
+      false // It doesn't make sense to paginate a single result, as pagination metadata would be discarded
```

Comment:
Now thinking about it, if we use `getByID` to get sequences as well, pagination might be requried

Reply:
Sequences are naturally paginated if we support filtering the sequence by sequence keys. If you have a list of items and you need the next "page" you only need to request objects where `sequenceKey` > `biggestKnownSequenceKey`